en/use-auto-failover.md

```diff
@@ -46,23 +29,67 @@ controllerManager:
+
+TiDB Operator collects the Pod health status by accessing the `/status` interface of each TiDB Pod and records the status in the `.status.tidb.members` field of the TidbCluster CR.
+
+Take a TiDB cluster with 3 Pods as an example. If a Pod fails for more than 5 minutes (`tidbFailoverPeriod` is configurable), TiDB Operator automatically does the following operations:
+
+1. TiDB Operator records the Pod information in the `.status.tidb.failureMembers` field of TidbCluster CR. 
+2. When calculating the replicas of TiDB StatefulSet, TiDB Operator takes the `.status.tidb.failureMembers` into account, so it will create a new Pod. Then, 4 Pods will exist at the same time.
+
+When the failed Pod in the cluster recovers, TiDB Operator will automatically scale in the newly created Pod, and the number of Pods gets back to 3.
+
+> **Note:**
+>
+> For each TiDB cluster, the maximum number of Pods that TiDB Operator can create is `spec.tidb.maxFailoverCount` (the default value is `3`). After the threshold is reached, TiDB Operator will not perform failover.
 
 ### Failover with TiKV
 
-When a TiKV Pod fails, its status turns to `Disconnected`. After 30 minutes (configurable by setting [`max-store-down-time`](https://pingcap.com/docs/stable/pd-configuration-file/#max-store-down-time) to `"30m"` in PD's configuration file), the status becomes `Down`. After waiting for another 5 minutes (configurable by modifying `tikvFailoverPeriod`), if this TiKV Pod is still down, TiDB Operator creates a new TiKV Pod.
+TiDB Operator collects the TiKV store health status by accessing the PD API and records the status in the `.status.tikv.stores` field in TidbCluster CR.
 
-If the failed TiKV Pod gets back online, TiDB Operator does not automatically delete the newly created Pod. This is because scaling in the TiKV Pods will trigger data transfer.
+Take a TiKV cluster with 3 Pods as an example. When a TiKV Pod fails, the store status of the Pod changes to `Disconnected`. By default, after 30 minutes (configurable by changing `max-store-down-time = "30m"` in the `[schedule]` section of `pd.config`), the status changes to `Down`. Then, TiDB Operator automatically does the following operations:
 
-If **all** of the failed Pods have recovered, and you want to scale in the newly created Pods, you can follow the procedure below:
+1. Wait for another 5 minutes (configurable by modifying `tikvFailoverPeriod`), if this TiKV Pod is still not recovered, TiDB Operator records the Pod information in the `.status.tikv.failureStores` field of TidbCluster CR.
+2. When calculating the replicas of TiKV StatefulSet, TiDB Operator takes the `.status.tikv.failureStores` into account, so it will create a new Pod. Then, 4 Pods will exist at the same time.
+
+When the failed Pod in the cluster recovers, TiDB Operator **DOES NOT** scale in the newly created Pod, but continues to keep 4 Pods. This is because scaling in TiKV Pods will trigger data migration, which might affect the cluster performance.
```

Comment:
```suggestion
When the failed Pod in the cluster recovers, TiDB Operator **DOES NOT** remove the newly created Pod, but continues to keep 4 Pods. This is because scaling in TiKV Pods will trigger data migration, which might affect the cluster performance.
```